### DISCUSSION TITLE
Changes in resources not represented in built app

### DISCUSSION THREAD

**Discussion by fwolter** [NightscoutFoundation/xDrip#2741] [Q&A] [answered]
Whenever I change a layout or a value in strings.xml for example, I need to delete `~/.gradle/caches/build-cache-1 ` and rebuild the project in Android Studio to make the change appear in the app in the emulator or on the physical device.

I tried Android Studio 2022.1.1.21 and 4.2.1. Both show the same behavior.

The behavior is also the same when building purely on the command line. It takes a `./gradlew cleanbuildcache clean` and an additional deletion of above cache directory. Installing the app with `./gradlew installFastDebug` shows the change, then.

Changes in code are reflected without deleting any cache or rebuilding anything. Do you have any hint, what could cause this?

**Comment by Navid200** [comment, marked as answer]
Every time you want to build, just go to "Build" on the title bar and do "Clean".  Then, do "Rebuild project".  Then, create the app.
It does make it a little bit longer.   But, only a minute or two.
I don't know any faster way.

**Comment by fwolter** [comment]
Thanks for the hint, that's indeed a bit easier! However, it didn't come to my mind this is normal behavior, so I spent two days on it :) And I thought rebuild does an implicit clean, now I know better. Thanks again!

**Comment by fwolter** [comment]
Just for the records, if you modify `insulin_profiles.txt`, you need to also remove `~/.gradle/caches/build-cache-1`. Running a clean/rebuild is not enough.

@jamorham Do you have an idea what could cause this?
